### PR TITLE
"shifted softplus added"

### PR DESCRIPTION
**Pull request by mayank-root**

### CODE REVIEW

File: src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SHIFTED_SOFTPLUS_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SHIFTED_SOFTPLUS_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+
+/**
+ * The Shifted Softplus function, defined by
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp:20]: Can we cite a paper as a reference?

File: src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ public:
+  /**
+   * Computes the Shifted-Softplus function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+      return std::log(0.5 * std::exp(x) + 0.5);
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp:38]: Can you check that the indentation matches with the rest of the codebase?

File: src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    template<typename InputVecType, typename OutputVecType>
+    static void Deriv(const InputVecType& y, OutputVecType& x)
+    {
+      x.set_size(y.size());
+      for(size_t i = 0; i < y.n_elem; ++i)
+      {
+          x(i) = (0.5 * std::exp(y(i))) / (0.5 * std::exp(y(i)) + 0.5);
+      }
+    }
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp:82]: See comment above.

File: src/mlpack/tests/ann/activation_functions_test.cpp [hunk 4]
```diff
@@ -805,4 +805,22 @@ TEST_CASE("HyperSinhFunctionTest", "[ActivationFunctionsTest]")
 
   CheckActivationCorrect<HyperSinhFunction>(activationData, desiredActivation);
   CheckDerivativeCorrect<HyperSinhFunction>(desiredActivation, desiredDerivate);
+}
+/**
+ * Basic test of the Shifted-Softplus Function
+ */
+TEST_CASE("ShiftedSoftplusTest", "[ActivationFunctionsTest]")
```
Comment (zoq) [inline src/mlpack/tests/ann/activation_functions_test.cpp:812]: Can you add the same tests as for the other activation functions?

Comment (mayank-root) [reply src/mlpack/tests/ann/activation_functions_test.cpp:812]: @zoq what do you mean by same tests as for the other activation functions?

Comment (mayank-root) [reply src/mlpack/tests/ann/activation_functions_test.cpp:868]: Do i need to add the jacobian test as well??

Comment (zoq) [reply src/mlpack/tests/ann/activation_functions_test.cpp:868]: correct

File: src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    template<typename InputVecType, typename OutputVecType>
+    static void Deriv(const InputVecType& y, OutputVecType& x)
+    {
+      x.set_size(y.size());
+      for(size_t i = 0; i < y.n_elem; ++i)
+      {
+        x(i) = (0.5 * std::exp(y(i))) / (0.5 * std::exp(y(i)) + 0.5);
+      }
+    }
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/shifted_softplus.hpp:93]:
```suggestion
class ShiftedSoftplusFunction
{
 public:
  /**
   * Computes the Shifted-Softplus function.
   *
   * @param x Input data.
   * @return f(x).
   */
  static double Fn(const double x)
  {
    return std::log(0.5 * std::exp(x) + 0.5);
  }

  /**
   * Computes the Shifted-Softplus function.
   *
   * @param x Input data.
   * @param y The resulting output activation.
   */
  template<typename InputVecType, typename OutputVecType>
  static void Fn(const InputVecType& x, OutputVecType& y)
  {
    y.set_size(x.size());
    for(size_t i = 0; i < x.n_elem; ++i)
    {
      y(i) = std::log(0.5 * std::exp(x(i)) + 0.5);
    }
  }

  /**
   * Computes the first derivative of the Shifted-Softplus function.
   *
   * @param y Input activation.
   * @return f'(x)
   */
  static double Deriv(const double y)
  {
    return (0.5 * std::exp(y)) / (0.5 * std::exp(y) + 0.5);
   }

  /**
   * Computes the first derivatives of the Shifted-Softplus function.
   *
   * @param y Input activations.
   * @param x The resulting derivatives.
   */
  template<typename InputVecType, typename OutputVecType>
  static void Deriv(const InputVecType& y, OutputVecType& x)
  {
    x.set_size(y.size());
    for(size_t i = 0; i < y.n_elem; ++i)
    {
      x(i) = (0.5 * std::exp(y(i))) / (0.5 * std::exp(y(i)) + 0.5);
     }
  }
```


Indentation fixes.

**Pull request closed without merging**